client/coral-embed/index.js

```diff
@@ -0,0 +1,163 @@
+        name: `${el.id  }_` + 'iframe'
+      }
+    );
+
+    configurePymParent(pymParent, asset);
+  };
+
+  return Coral;
+
+  // build the URL to load in the pym iframe
+  function buildStreamIframeUrl(talkBaseUrl, asset) {
+    let iframeUrl = [
+      talkBaseUrl,
+      (talkBaseUrl.match(/\/$/) ? '' : '/'), // make sure no double-'/' if opts.talk already ends with '/'
+      'embed/stream?asset_url=',
+      encodeURIComponent(asset)
+    ].join('');
+    return iframeUrl;
+  }
+
+  // Set up postMessage listeners/handlers on the pymParent
+  // e.g. to resize the iframe, and navigate the host page
+  function configurePymParent(pymParent, assetUrl) {
+    let notificationOffset = 200;
+    let ready = false;
+
+    // Resize parent iframe height when child height changes
+    pymParent.onMessage('height', function(height) {
+
+      // TODO: In local testing, this is firing nonstop. Maybe there's a bug on the inside?
+      // Or it's by design of pym... but that's very wasteful of CPU and DOM reflows (jank)
+      pymParent.el.querySelector('iframe').height = `${height  }px`;
+    });
+
+    // Helps child show notifications at the right scrollTop
+    pymParent.onMessage('getPosition', function() {
+      let position = viewport().height + document.body.scrollTop;
+
+      if (position > notificationOffset) {
+        position = position - notificationOffset;
+      }
+
+      pymParent.sendMessage('position', position);
+    });
+
+    // Tell child when parent's DOMContentLoaded
+    pymParent.onMessage('childReady', function () {
+      var interval = setInterval(function () {
+        if (ready) {
+          window.clearInterval(interval);
+
+          // @todo - It's weird to me that this is sent here in addition to the iframe URL. Could it just be in one place?
+          pymParent.sendMessage('DOMContentLoaded', assetUrl);
+        }
+      }, 100);
+    });
+
+    // When end-user clicks link in iframe, open it in parent context
+    pymParent.onMessage('navigate', function (url) {
+      window.open(url, '_blank').focus();
+    });
+
+    // wait till images and other iframes are loaded before scrolling the page.
+    // or do we want to be more aggressive and scroll when we hit DOM ready?
+    document.addEventListener('DOMContentLoaded', function () {
+      ready = true;
+    });
+
+    // get dimensions of viewport
+    function viewport() {
+      let e = window, a = 'inner';
+      if ( !( 'innerWidth' in window ) ){
+        a = 'client';
+        e = document.documentElement || document.body;
+      }
+      return {width : e[ `${a}Width` ], height : e[ `${a}Height` ]};
+    }
+  }
+
+  // return a reference to pym.js
+  function requirePym() {
+    let pym;
+
+    // fake AMD `define` so that the pym.js copypasta doesn't create a global
+    function define(createPym) {
+      pym = createPym();
+    }
+    define.amd = true;
+
+    /* eslint-disable */
+    /* ! pym.js - v1.1.2 - 2016-10-25 */
```

Comment:
We should be able to add a step to webpack to build in licenses into the embed file.